host.py

```diff
@@ -296,20 +296,43 @@ def copy_to(self, src_file: str, dst_file: str) -> None:
     def need_sudo(self) -> None:
         self.sudo_needed = True
 
-    def run(self, cmd: str, log_level: int = logging.DEBUG, env: Dict[str, str] = os.environ.copy()) -> Result:
+    def run(
```

Comment:
The current lines are only so long because of `line-length = 250` in `pyproject.toml`. That is at least a questionable choice when the experts (peop8 and python-black authors who do nothing but think about the optimal Python style) decide that a line length of 80 (pep8) or 88 (black) is best. IMO I also think that long lines hurt readability (especially with type annotations) and having one argument per line is preferable. But style is of course subjective. IMO black configuration from `pyproject.toml` should be removed, and then black would wrap those lines.

Anyway. I don't really care.

Reply:
Do you mean, not wrapping the line?

At the end of the branch, that would result in a 246 character long line. 
```python
    def run(self, cmd: Union[str, Iterable[str]], log_level: int = logging.DEBUG, env: Optional[Dict[str, str]] = None, *, env_extra: Optional[Mapping[str, Optional[str]]] = None, cwd: Optional[str] = None, sudo: Optional[bool] = None) -> Result:
```

The trailing comma to a list or function arguments (like `foo(a, b, )`) tells python-black to wrap the lines for each argument. Using that formatting feature, is hardly a violation of the existing style, unless the current style is never to add those trailing commas. It doesn't disagree with black formatting, because this is a feature of black. Requesting a line break from black is not a violation of the style. You can do it, if you think it results in nicer formatting.